src/utils/branded-types.ts

```diff
@@ -0,0 +1,15 @@
+declare const __brand: unique symbol;
+
+type Brand<B> = { [__brand]: B };
+
+export type Branded<T, B> = T & Brand<B>;
+
+export type AuthCode = Branded<string, 'AuthCode'>;
+
+export type Token = Branded<string, 'Token'>;
+
+export type ClientID = Branded<string, 'ClientID'>;
+
+export type ClientSecret = Branded<string, 'ClientSecret'>;
+
+export type HostName = Branded<string, 'HostName'>;
```

Comment:
Could we please name this `Hostname` vs `HostName`

Reply:
Done 🙏🏾